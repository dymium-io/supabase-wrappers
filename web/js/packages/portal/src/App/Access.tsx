import React, { useEffect, useState, useRef } from 'react';
import Tabs from 'react-bootstrap/Tabs'
import Tab from 'react-bootstrap/Tab'
import { useLocation, useNavigate } from "react-router-dom";
import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import { Link } from "react-router-dom";
import Alert from 'react-bootstrap/Alert'
import Spinner from '@dymium/common/Components/Spinner'
import * as http from '@dymium/common/Api/Http'
import * as com from '../Common'
import * as types from '@dymium/common/Types/Common'
import * as internal from '@dymium/common/Types/Internal'
import { useInitialize } from '../Utils/CustomHooks'
import { useAppDispatch, useAppSelector } from './hooks'
import { setActiveAccessTab } from '../Slices/menuSlice'

function YourDatascopes() {
  const [spinner, setSpinner] = useState(false)
  const [alert, setAlert] = useState<JSX.Element>(<></>)
  const [datascopes, setDatascopes] = useState<types.UserDatascopes>()
  let port = com.getTokenProperty("port")
  //"/api/getdatascopesaccess"
  let displayDatascope = (index) => {

  }
  let commandLine = (scope) => {
    let c = `psql -h localhost -p ${port} -d ${scope.name} -U ${datascopes !== undefined && datascopes.username}`

    return <div style={{ display: "flex" }}> <div className=" terminal">
      &gt;{c}
      </div>
      <i onClick={copy(c)} className="fas fa-copy clipbtn"></i>
    </div>
  }
  let python = scope => {
    let p = `conn = psycopg2.connect("host=localhost port=${port} dbname=${scope.name} user=${datascopes !== undefined && datascopes.username} password=${datascopes !== undefined && datascopes.password}")`

  return <div style={{ display: "flex" }}> <div className=" terminal">
        {p}
      </div>
      <i onClick={copy(p)} className="fas fa-copy clipbtn"></i>
    </div>
  }
let java = scope => {
  let j = `
  String url = "jdbc:postgresql://localhost:`+ port + `/`+scope.name+`";
  Properties props = new Properties();
  props.setProperty("user","${datascopes !== undefined && datascopes.username}");
  props.setProperty("password","${datascopes !== undefined && datascopes.password}");
  
  `

return <div style={{ display: "flex" }}> 
  <div><div className=" terminal"> <pre><code>{j}
  </code></pre></div>
  </div> 
  <i onClick={copy(j)} className="fas fa-copy clipbtn"></i>  
  </div>
}
let displayDatascopes = () => {
  let copy = () => {
    return e => {
      if (datascopes !== undefined)
        navigator.clipboard.writeText(datascopes.password);
    }
  }

  return <div>
    <div className="viewport">
      <div className="thickblue mb-2" style={{ fontWeight: 'bold' }}><i className="fa fa-key mr-2" aria-hidden="true"></i>Credentials:</div>
      <div>Username: {datascopes !== undefined && datascopes.username}</div>
      <div>Password: {datascopes !== undefined && datascopes.password} 
      <Button onClick={regenerate} style={{ marginTop: '-4px', paddingBottom: '0.1em', paddingTop: '0.0em' }} variant="dymium" className="mx-2" size="sm">Regenerate</Button> 

       {datascopes !== undefined && datascopes.password !== `**********` &&
        
     
        <i onClick={copy()}  className="fas fa-copy clipbtn"></i>

      } </div>
      <div className="mt-3" style={{ fontSize: '0.8em' }}>Password is only visible to you for a day. You can always generate a new one.</div>
    </div>
    {datascopes !== undefined && datascopes.datascopes.map(x => {
      return <div className="my-5">
        <h5 className="mb-2"><i className="fa fa-unlock mr-2" aria-hidden="true"></i>Ghost Database: {x.name}</h5>
        <div className="datascopeuse">
          <Tabs
            id="datascope"
            unmountOnExit={true} className=" mt-0 text-left ">
            <Tab eventKey="cmd" title="Command Line" className=" mx-2">
              {commandLine(x)}
            </Tab>
            <Tab eventKey="python" title="Python" className=" mx-2">
              {python(x)}
            </Tab>
            <Tab eventKey="java" title="Java" className=" mx-2">
              {java(x)}
            </Tab>

          </Tabs>
        </div>
      </div>
    })}
  </div>
}
let getDatascopeAccess = () => {
  let error = <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>
    Error retrieving datascopes.
  </Alert>
  setSpinner(true)
  http.sendToServer("GET", "/api/getdatascopesaccess",
    null, "",
    resp => {
      resp.json().then(js => {
        setDatascopes(types.UserDatascopes.fromJson(js))
        setSpinner(false)
      }).catch((error) => {
        setAlert(
          <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error.message}</Alert>                  
        )
        setSpinner(false)
      })
    },
    resp => {
      setSpinner(false)
      setAlert(
        <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>Get datascope access failed</Alert>                  

      )
      setSpinner(false)
    },
    error => {
      console.log("on exception")
      setSpinner(false)
      setAlert(
        <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error.message}</Alert>                  

      )
      setSpinner(false)
    })
}

let regenerate = () => {
  let error = <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>
    Error regenerating password.
  </Alert>
  setSpinner(true)
  http.sendToServer("GET", "/api/regenpassword",
    null, "",
    resp => {
      resp.json().then(js => {
        setDatascopes(types.UserDatascopes.fromJson(js))
        setSpinner(false)
      }).catch((error) => {
        setAlert(
          <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error.message}</Alert>                  
        )
        setSpinner(false)
      })
    },
    resp => {
      setSpinner(false)
      setAlert(  <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>"Regenerate password failed"</Alert>        
      )
      setSpinner(false)
    },
    error => {
      console.log("on exception")
      setSpinner(false)
      setAlert(
        <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error.message}</Alert>        
      )
      setSpinner(false)
    })
}

useEffect(() => {
  getDatascopeAccess()
}, [])

return (
  <div className=" text-left">
    {alert}
    <h5 > Your Ghost Databases <Spinner show={spinner} style={{ width: '28px' }}></Spinner></h5>
    <div className=" text-left">

      {datascopes !== undefined && displayDatascopes()}


    </div>
  </div>
)
}
let orgid = com.getTokenProperty("schema")
let url = window.location.protocol + "//" + window.location.host + "/"
let params = `-c ${orgid} -p ${url}`

function copy(str) {

  return e => {
    navigator.clipboard.writeText(str);
  }
}
function Downloads() {

  return (
    <div className=" text-left">

      <h5 > Client Downloads</h5>
      <div className=" text-left">

        <div className="viewport">
          <div>On MS Windows:</div>
          <a href="/DymiumInstaller.exe" download="DymiumInstaller.exe"> <i className="fab fa-windows mr-2" aria-hidden="true"></i>Click to Download MS Windows Client</a>
          <div>
            CLI Usage:
            <div style={{ display: "flex" }}>
              <div className="terminal">
                &gt;dymium.exe {params}
              </div>
              <i onClick={copy("dymium.exe " + params)} className="fas fa-copy clipbtn"></i>
            </div>
          </div>
        </div>


        <div className="viewport">
          <div>On Mac OS X:</div>
          <a href="/DymiumInstaller.pkg" download="DymiumInstaller.pkg"> <i className="fab fa-apple mr-2" aria-hidden="true"></i>Click to Download Mac OS X Client</a>
          <div>
            CLI Usage:
            <div style={{ display: "flex" }}>
              <div className="terminal">
                &gt;./dymium {params}
              </div>
              <i onClick={copy("./dymium " + params)} className="fas fa-copy clipbtn"></i>
            </div>
          </div>
        </div>

        <div className="viewport">
          <div>On Linux:</div>
          <a href="/tunnel.tar.gz" download="tunnel.tar.gz"> <i className="fab fa-linux mr-2" aria-hidden="true"></i>Click to Download Linux Client</a>
          <div>
            CLI Usage:
            <div style={{ display: "flex" }}>
              <div className="terminal">
                &gt;./dymium {params}
              </div>
              <i onClick={copy("./dymium " + params)} className="fas fa-copy clipbtn"></i>
            </div>
          </div>
        </div>
        <div>
        To use the GUI, launch dymiumgui.app on Mac OS, or dymiumgui.exe on Windows.
        </div><div>
        <img src="/dymiumgui.png" style={{width: "540px"}}></img>  
        <div  className="mb-5">
        Paste the connection string for your OS into the application, and click Connect.
        </div>
        </div>
      </div>
    </div>
  )
}
function useQuery() {
  const { search } = useLocation();

  return React.useMemo(() => new URLSearchParams(search), [search]);
}
export default function Access() {
  const navigate = useNavigate();
  let t = useAppSelector((state) => {

    return state.reducer.activeAccessTab
  }
  )
  console.log("t=", t)
  const appDispatch = useAppDispatch()
  let query = useQuery();

  useEffect(() => {
    if (query.get("key") !== undefined) {

      appDispatch(setActiveAccessTab(query.get("key")))
      navigate("/app/access")
    }
  }, [])
  let tt = query.get("key")
  if (tt !== null) {
    t = tt
  }
  if (t == null) {
    t = "datascopes"
  }
  return (
    <Tabs defaultActiveKey={t}
      onSelect={(k) => appDispatch(setActiveAccessTab(k))}

      unmountOnExit={true} className="mb-3 text-left">
      <Tab eventKey="datascopes" title="Your Ghost Databases" className="mx-4">
        <YourDatascopes />
      </Tab>
      <Tab eventKey="download" title="Client Download And Use" className="mx-4">
        <Downloads />
      </Tab>

    </Tabs>
  )
}