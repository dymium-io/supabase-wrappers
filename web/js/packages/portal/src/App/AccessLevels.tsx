import React, { Component, useEffect, useState, useRef } from 'react';

import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import { SortableContainer, SortableElement, SortableContainerProps, SortableElementProps, arrayMove } from 'react-sortable-hoc';

import Offcanvas from '@dymium/common/Components/Offcanvas'

import 'react-bootstrap-table-next/dist/react-bootstrap-table2.min.css';
import Spinner from '@dymium/common/Components/Spinner'
import Alert from 'react-bootstrap/Alert'
import * as types from '@dymium/common/Types/Common'
import * as http from '@dymium/common/Api/Http'
import * as com from '../Common'

import { defaultDetectors, regexpDetectors } from "./Detectors"


let handlingOptions = () => {
  let opts: types.DataHandling[] = ["allow", "block", "obfuscate", "redact"]

  return opts.map(x => {
    return <option key={x} value={x}>
      {types.humanReadableDataHandling(x)}
    </option>
  })
}

export default function AccessLevels() {
  const [spinner, setSpinner] = useState(false)
  const [alert, setAlert] = useState<JSX.Element>(<></>)

  const [name, setName] = useState("")
  const [level, setLevel] = useState("")
  const [actions, setActions] = useState<types.DataAction[]>([])
  const [showOffhelp, setShowOffhelp] = useState(com.isInstaller())

  let policy = useRef(new types.DataPolicy())
  let errorGet = <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>
    Error retrieving policy.
  </Alert>
  let savePolicies = () => {

    for (let i = 0; i < actions.length; i++) {
      actions[i].index = i
    }
    policy.current.actions = actions

    let js = JSON.parse(policy.current.toJson())
    // do cleannup
    for (let i = 0; i < js.piisuggestions.length; i++) {
      let sactions = js.piisuggestions[i].actions
      let gactions = js.actions
      let out: types.DataAction[] = []
      let getAction = l => {
        for (let i = 0; i < sactions.length; i++) {
          if (sactions[i].role === l.role)
            return sactions[i]
        }
        return l
      }
      for (let j = 0; j < gactions.length; j++) {
        let l = getAction(gactions[j])
        out.push(l)
      }
      js.piisuggestions[i].actions = out

    }
    let body = JSON.stringify(js)
    setSpinner(true)
    http.sendToServer("POST", "/api/savepolicies",
      null, body,
      resp => {
        resp.json().then(js => {
          setSpinner(false)
          if (js.status !== "OK") {

            getPolicies()
            setAlert(
              <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{js.error}</Alert>
            )
          } else {
            setAlert(
              <Alert variant="success" onClose={() => setAlert(<></>)} dismissible>Access levels saved successfully</Alert>
            )
            policy.current = types.DataPolicy.fromJson(js)

          }
        }).catch((error) => {
          setAlert(
            <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error}</Alert>
          )
          setSpinner(false)
        })
      },
      resp => {
        setSpinner(false)
        resp != null && resp.text().then(t =>
          setAlert(
            <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>Error retrieving policy: {t}</Alert>
          ))
        setSpinner(false)
      },
      error => {
        console.log("on exception")
        setSpinner(false)
        setAlert(
          <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error.message}</Alert>

        )
        setSpinner(false)
      })
  }
  let initializePolicy = () => {
    policy.current = new types.DataPolicy()

    let det = [...defaultDetectors]
    let newa = det.concat(regexpDetectors)

    let prep = newa.map(x => {
      let y = types.PIISuggestor.fromJson({ actions: [], detector: x.detector })
      return y
    })
    policy.current.piisuggestions = prep
  }
  let getPolicies = () => {
    let error = <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>
      Error retrieving policy.
    </Alert>
    setSpinner(true)
    http.sendToServer("GET", "/api/getpolicies",
      null, "",
      resp => {
        resp.json().then(js => {
          setSpinner(false)

          if (js.error !== undefined) {
            // need to initialize the whole iguana
            initializePolicy()

          } else {
            let prep = types.DataPolicy.fromJson(js)
            if (js.piisuggestions.length === 0) {
              initializePolicy()
            } else {
              policy.current = prep
            }
            setActions(prep.actions)
          }
        }).catch((error) => {
          setAlert(
            <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error.message}</Alert>

          )
          setSpinner(false)
        })
      },
      resp => {
        setSpinner(false)
        resp != null && resp.text().then(t =>
          setAlert(
            <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>getpolicies failed: {t}</Alert>

          ))
        setSpinner(false)
      },
      error => {
        console.log("on exception")
        setAlert(
          <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error.message}</Alert>
        )
        setSpinner(false)
      })
  }

  useEffect(() => {
    getPolicies()
  }, [])

  let onDelete = index => {
    return e => {
      let v = actions.splice(index, 1)
      setActions(v)
    }
  }
  const SortableItem: React.ComponentClass<SortableElementProps & { value: string, iindex: number }, any> = SortableElement(({ value, iindex }: { value: string, iindex: number }) => {
    let onDelete = e => {
      actions.splice(iindex, 1)
      setActions([...actions])
    }
    let onChange = e => {
      actions[iindex].handling = e.target.value
      setActions([...actions])
    }
    return <li className="card licard">
      <Row>
        <Col><div style={{ marginTop: '4px' }}>
          {value}</div>
        </Col>
        <Col xs="auto">

          <select onChange={onChange} style={{ marginTop: '4px' }} value={actions[iindex].handling} className="form-control form-control-sm">
            {handlingOptions()}
          </select>
        </Col>
        <Col xs="auto">
          <Button variant="outline" onClick={onDelete} ><i className="fas fa-trash ablue" aria-label={"delete" + iindex} id={"delete" + iindex} ></i></Button>
        </Col>
      </Row>

    </li>
  }
  );

  const SortableList: React.ComponentClass<SortableContainerProps & { items: types.DataAction[] }, any> = SortableContainer(({ items }: { items: types.DataAction[] }) => {
    return (
      <ul style={{ listStyle: 'none', width: '50%' }} className="liouter">
        {items.map((value: types.DataAction, index: number) => (
          <SortableItem key={`item-${index}`} index={index} iindex={index} value={value.role} />
        ))}
      </ul>
    );
  });


  let addLevel = event => {
    //let x: types.DataHandling = 'allow <*> Allow'
    let v: types.DataAction = new types.DataAction
    v.role = name

    switch (level) {
      case 'allow':
        v.handling = 'allow'
        break
      case 'block':
        v.handling = 'block'
        break
      case 'obfuscate':
        v.handling = 'obfuscate'
        break
      case 'redact':
        v.handling = 'redact'
        break
      default:
        v.handling = 'block'
        break
    }
    actions.push(v)
    setActions([...actions])

    event.preventDefault();
    //setValidated(false)
    event.stopPropagation();
    setName("")
  }

  let onSortEnd = ({ oldIndex, newIndex }) => {
    var a = arrayMove(actions, oldIndex, newIndex)

    setActions(a)
  }

  let handleSubmit = event => {
    event.preventDefault();
    //setValidated(false)
    event.stopPropagation();
    savePolicies()
  }

  return <div>
    <h5 >Define Access Levels <i onClick={e => { setShowOffhelp(!showOffhelp) }} className="trash fa-solid fa-circle-info mr-1"></i> <Spinner show={spinner} style={{ width: '28px' }}></Spinner></h5>
    <Offcanvas modal={false} width={300} show={showOffhelp} onClose={(e) => { setShowOffhelp(false) }}>
      <h5>Access Levels</h5>
      <div className="mb-3">
        Access levels are the templates for access rights when you define the Ghost Databases. For example, you can define levels
        <ul>
          <li>
            Administrator
          </li>
          <li>
            Data Engineer
          </li>
          <li>
            Data Scientist
          </li>
        </ul>

      </div>
      <div className="mb-3">
        Once you defined the levels, go to the next tab Rules, and set the suggestion for data handling for various PIIs
      </div>



    </Offcanvas>
    {alert}
    <div>
      <Form onSubmit={addLevel}>
        <Row >
          <Col xs="auto">
            <Form.Group className="mb-3" controlId="dname">
              <Form.Label>Name:</Form.Label>
              <Form.Control size="sm" type="text" placeholder="alpha_num, small caps"
                required
                pattern="[\w '&%]+"
                value={name}
                onChange={e => setName(e.target.value)}
              />
              <Form.Control.Feedback >Looks good!</Form.Control.Feedback>
              <Form.Control.Feedback type="invalid" >
                Type systemwide unique name to use in SQL
              </Form.Control.Feedback>
            </Form.Group>
          </Col>
          <Col xs="auto">
            <Form.Group className="mb-3" controlId="daction">
              <Form.Label>Default Action:</Form.Label>
              <Form.Control as="select" required size="sm"
                value={level}
                onChange={e => {
                  setLevel(e.target.value)
                }}
              >
                {handlingOptions()}

              </Form.Control>
            </Form.Group>
          </Col>
          <Col xs="auto">
            <Button style={{ position: 'relative', top: '0.16em' }} variant="dymium" size="sm" className="mt-4" type="submit">Add</Button>
          </Col>
        </Row>
      </Form>
      <Form onSubmit={handleSubmit}>

        <>
          {actions.length > 0 &&
            <SortableList distance={1} items={actions} onSortEnd={onSortEnd} />
          }
          <Row className="mt-5">
            <Col xs="auto">
              <Button size="sm" variant="dymium" type="submit">Apply</Button>
            </Col>
          </Row>
        </>


      </Form>
    </div>
  </div>
}
