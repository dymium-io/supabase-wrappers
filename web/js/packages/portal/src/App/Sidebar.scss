/* Fixed sidenav, full height */

$primary-color: rgb(255, 158, 24);
$primary-color-text: rgb(31, 31, 32);
$primary-color-2: rgb(222, 139, 22);
$primary-color-2-text: #eee;
$primary-color-3: rgb(220, 158, 140); 
$primary-color-3-text: black;
$primary-pale: rgb(183, 114, 17);
$primary-pale-text: #eeeeee;
$secondary-color: rgb(6, 33, 76); 

.sidenav {
    height: 100% !important;
    
    width: 160px;
    position: sticky;
    z-index: 1;
    top: 0;
    left: 0;
    background-color: $primary-color;
    color: $primary-color-text ;
    overflow-x: hidden;
    padding-top: 13px;
  }
  .navselected {
    background-color: $primary-color-2 !important;
    color: $primary-color-2-text !important;
  }
  .sidesection {
    padding: 4px 4px 14px 30px;
    margin-left : 3px;
    text-align: left;
    color: rgba(255,255,255,1);
  }
  /* Style the sidenav links and the dropdown button */
  .sidenav a, .side-dropdown-btn {
    padding: 5px 7px 5px 8px;
    text-decoration: none;
    font-size: 17px;
  
    display: block;
    border: none;
    background: none;
    width:100%;
    text-align: left;
    cursor: pointer;
    outline: none;
    border-bottom: 0.5px solid $primary-color-3;
   
  }
.side-dropdown-btn {
  padding: 6px 2px 6px 8px
}  
.fa-caret-down {
  margin-top: 0px !important;
}
.green  {
  color: rgb(155, 237, 176);
}
.sidenav .notalink {
  padding: 8px 13px 8px 25px;
  text-decoration: none;
  font-size: 17px;
  color: rgb(235, 235, 235);
  display: block;
  border: none;
  background: none;
  width:100%;
  text-align: left;
  cursor: not-allowed;
  outline: none;
}

/* On mouse-over */
.sidenav  a:hover, .side-dropdown-btn:hover {
  color: $primary-pale-text !important;
  background-color: $primary-pale !important;
}



/* Dropdown container (hidden by default). Optional: add a lighter background color and some left padding to change the design of the dropdown content */
.dropdown-container, .side-dropdown-btn:focus {
  display: none;
  background-color: $primary-color-2;
  color: $primary-color-2-text !important;
  outline: none;    
  border: none;    
}
.selectpad {
  margin-top: '2px' !important;
  padding-left: 0px !important;  
  padding-right: 28px !important;  

}
.selectsign {
  background-color: $primary-pale !important;
  
  padding-left: 0px !important;  
  padding-right: 8px !important;  
}
.selectsigntransparent {
    
  padding-left: 0px !important;  
  padding-right: 8px !important;  
}
.selectionoffset {
padding-left: 0px !important;
padding-top: 0px !important;
padding-bottom: 0px !important;
margin-top: 0px !important;
margin-bottom: 0px !important;
}
/* Optional: Style the caret down icon */
.fa-caret-down {
  float: right;
  padding-right: 8px;    
}
.sidesection strong{
  color: white !important;
}
.sidebartext {
  font-size: 13px !important;
  margin-left: 0.3em;
  color: rgb(243, 227, 205);
}
