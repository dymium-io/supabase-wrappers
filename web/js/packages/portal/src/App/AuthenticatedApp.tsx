import React, { useEffect } from 'react';
import { Outlet } from "react-router-dom";
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Backdrop from "../Backdrop"
import Menu from './Menu'
import Auth from '../Auth'
import { Link } from "react-router-dom";
import Sidebar from './Sidebar'
import * as com from '../Common'

const AuthenticatedApp = () => {
  let roles = com.getTokenProperty("roles")
  if (roles === null) {
    return <div className="py-0 my-0">
      <Backdrop />
      <div className="text-center" style={{
        position: 'absolute', top: '0px', left: '0px',
        width: '100%', height: '100vh'
      }}>
        <div id="loginbox" >
          <div className="row">
            <div className="col-sm ml-5  text-center">
              <h1 style={{ marginTop: '1.3em', fontSize: '5em', fontWeight: '300' }} className="logoheader ">
                <Link className="linklink" to="/" > Dymium </Link></h1>
            </div>
            <div className="col-sm mt-5  text-center">
              <h1 className="mt-5 logoheader">Error</h1>
              <h1 className="unauthorized" >Not authorized!</h1>
              <div className="pt-2 logofooter">
                You are succesfully authenticated but not authorized to use Dymium. Please contact your administrator.
              </div>

            </div>
          </div>

        </div>
      </div>



    </div>
  }
  return (
    <>
      <Auth />
      <Menu />

      <Row style={{ height: '100vh' }} className="p-0 m-0">
        <Col xs="auto" id="sidebar_container" className="px-0 sidebar_container"><Sidebar /></Col>
        <Col xs className="p-0 spider" style={{ overflowX: "scroll" }}>
          <Outlet />
        </Col>

      </Row>

    </>
  )
}

export default AuthenticatedApp;