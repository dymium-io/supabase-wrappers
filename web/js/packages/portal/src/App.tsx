import React, { useEffect } from 'react';
import LandingPage from './LandingPage'
import AuthenticatedApp from './App/AuthenticatedApp'
import Error404 from "./Error404"
import Logout from "./Logout"
import Error from "./Error"
import Invitation from "./Invitation"
import Sticky from "./App/Sticky"
import Dashboard from "./App/Dashboard"
import Customers from "./App/Customers"
import Connections from "./App/Connections"
import Datascopes from "./App/Datascopes"
import Groups from "./App/Groups"
import Rules from "./App/Rules"
import TestSQL from "./App/TestSQL"
import * as com from './Common'

import "@fontsource/roboto";
import 'bootstrap/dist/css/bootstrap.min.css';
import 'react-bootstrap-typeahead/css/Typeahead.css';

import { BrowserRouter, Routes, Route } from "react-router-dom";
//import Helmet from "react-helmet";

import '@dymium/common/App.scss';


function App() {
  return (
    <div className="App" id="app">

      <BrowserRouter>
        <Routes>

          <Route  path="/" element={<LandingPage />} />

          <Route path="/app/" element={<AuthenticatedApp />} >
            <Route path="/app/dashboard" element={<Dashboard />} />
            <Route path="/app/connections" element={<Connections />} />     
            <Route path="/app/datascopes" element={<Datascopes />} />                
            <Route path="/app/groups" element={<Groups />} />    
            <Route path="/app/rules" element={<Rules />} />         
            <Route path="/app/test" element={<TestSQL />} />                                             
            <Route path="/app/" element={<Sticky />} />          
          </Route>
          <Route  path="/app/logout" element={<Logout/>} />
          <Route path="/app/error" element={<Error />} />
          <Route path="/login" element={<Invitation />} />
          <Route path="*" element={<Error404 />} />
        </Routes>
      </BrowserRouter>

    </div>
  );
}

export default App;
