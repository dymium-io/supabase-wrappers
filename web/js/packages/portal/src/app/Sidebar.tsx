import React, { useState } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import '@dymium/common/App.css';
import { Link } from 'react-router-dom'
import * as com from '../Common'
import { useAppDispatch, useAppSelector } from './hooks'
import { setActiveMenu } from '../Slices/menuSlice'


import './Sidebar.css';




com.customizeStyleSheet()
export default function Sidebar() {
  const selected = useAppSelector((state) => {

    return state.reducer.activeMenu
  })
  const appDispatch = useAppDispatch()

  let getMenuItems = () => {
    let items: any[] = []
    items.push(
      {
        item: <div className='darkblue'> <i className="fas fa-tachometer-alt mr-1 fa-fw "></i>Dashboard </div>,
        to: '/app/dashboard',
        id: 'dashboard'
      },
      {
        item: <div className='darkblue'> <i className="fas fa-database mr-1 fa-fw"></i>Connections</div>,
        to: '/app/connections',
        id: 'connections'
      },
      {
        item:  <div className='darkblue'><i className="fas fa-lock mr-1  fa-fw"></i>Data Scopes</div>,
        to: '/app/datascopes',
        id: 'datascopes'
      },      
      {
        item:  <div className='darkblue'>  <i className="fas fa-users mr-1 fa-fw"></i>Groups </div>,
        to: '/app/groups',
        id: 'groups'
      },
      {
        item: <div className='darkblue'> <i className="fa fa-gavel mr-1 fa-fw"></i>Rules </div>,
        to: '/app/rules',
        id: 'rules'
      },      
      {
        item:  <div className='darkblue'>  <i className="fa-solid fa-vial mr-1 fa-fw"></i>Test SQL </div>,
        to: '/app/test',
        id: 'test'
      },      
  
    )

    return items.map(ob => {
      let cl = 'hover-sidebar '
    
      if(ob.id === selected) {
        cl = 'hover-sidebar navselected '
      }
      console.log(ob.id + ", " + selected + ", " + cl) 
      return <Link onClick={(k) => appDispatch( setActiveMenu(ob.id) )} key={ob.id} id={ob.id} className={cl} to={ob.to} > {ob.item} </Link>
    }
    )

  }
  return (
    <div className="sidenav h-100" id="sidebar">
      {getMenuItems()}

      <a  className='darkblue hover-sidebar '
        href="/api/logout"
      >
        <i className="fas fa-sign-out mr-1 fa-fw " aria-hidden="true"></i><span className="">Logout</span></a >
        </div>


  )
}

