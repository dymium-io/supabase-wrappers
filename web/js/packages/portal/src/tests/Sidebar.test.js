/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';

import { BrowserRouter, Routes, Route } from "react-router-dom";

import { Provider } from 'react-redux'
import { PersistGate } from 'redux-persist/integration/react';
import { persistStore } from 'redux-persist';
import store from '../Store/index'
import Sidebar from '../App/Sidebar'
import 'bootstrap/dist/css/bootstrap.min.css';
import '@dymium/common/App.scss';
import "@fortawesome/fontawesome-free/css/all.min.css";

let persistor = persistStore(store);




test('Instantiate sidebar', () => {
    const component = renderer.create(
        <Provider store={store}>
            <PersistGate loading={null} persistor={persistor}>
                <BrowserRouter>
                    <Sidebar />
                </BrowserRouter>
            </PersistGate>
        </Provider >
    );
    let tree = component.toJSON();
   
    expect(tree).toMatchSnapshot();
});