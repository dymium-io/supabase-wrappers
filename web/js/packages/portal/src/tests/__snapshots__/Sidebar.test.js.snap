// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instantiate sidebar 1`] = `
<div
  className="sidenav h-100"
  id="sidebar"
>
  <a
    className="hover-sidebar navselected "
    href="/app/access"
    id="access"
    onClick={[Function]}
  >
     
    <div
      className="darkblue"
    >
       
      <i
        className="fa fa-key mr-2 fa-fw"
      />
      User Access 
    </div>
     
  </a>
  <a
    className="hover-sidebar "
    href="/app/test"
    id="test"
    onClick={[Function]}
  >
     
    <div
      className="darkblue"
    >
        
      <i
        className="fa-solid fa-vial mr-2 fa-fw"
      />
      Test SQL 
    </div>
     
  </a>
  <a
    className="darkblue hover-sidebar "
    href="/api/logout"
  >
    <i
      aria-hidden="true"
      className="fas fa-sign-out mr-2 fa-fw "
    />
    <span
      className=""
    >
      Logout
    </span>
  </a>
</div>
`;
